Modules/CaPTkSurvival/src/CaPTkSurvivalPredictionAlgorithm.cpp

```diff
@@ -485,8 +498,8 @@ int SurvivalPredictionModuleAlgorithm::PrepareNewSurvivalPredictionModel(const s
   }
   catch (const std::exception& e1)
   {
-      MITK_ERROR << "Cannot find the file 'Survival_HMFeatures_Configuration.csv' in the ../data/survival directory. Error code : " + std::string(e1.what());
-      throw e1.what(); // propagate error message up to algorithm-running logic
+      MITK_ERROR << "Cannot find the file 'Survival_HMFeatures_Configuration.csv' in the ../models/survival_model directory.";
+      throw std::exception(e1); // propagate error message up to algorithm-running logic
   }
```

Comment:
Are you catching exceptions and propagating them because you don't want to mess with the legacy code too much? Because otherwise this is bad design.

At least a simpler way would be to raise your custom exception and do all the important catching/MITK_ERRORing from the calling method, like you do anyway. i.e. changing the above to this:

```cpp
  catch (const std::exception& e)
  {
      throw std::exception(
            std::string("Cannot find the file 'Survival_HMFeatures_Configuration.csv' in the ../models/survival_model directory: ") 
            + e.what()
      );
  }
```

Obviously the best solution would be to centralize all the different operations into functions or something. But this is generally fine if it is just for legacy reasons

Reply:
(maybe throw std::runtime_error instead of std::exception)